### PR TITLE
Add Visitor Typing using ChasitorTyping API

### LINKED ISSUE #8: [FEATURE] User Typing Indicators
- [ ] Sending 'ChasitorTyping' event when Live Chat user starts typing.

When user is typing in Livechat Widget show indicator in salesforce console

This is needed for https://github.com/RocketChat/Apps.Salesforce.LiveAgent/issues/10

### PR DESCRIPTION
**Pull request by Shailesh351**
## Related PRs :

RC Server: https://github.com/WideChat/Rocket.Chat/pull/422
Apps Engine: https://github.com/WideChat/Rocket.Chat.Apps-engine/pull/3

### CODE REVIEW

**Comment by Shailesh351** [comment]
@PrajvalRaval Your review will help here 😄

**Comment by PrajvalRaval** [comment]
> @PrajvalRaval Your review will help here 😄

Sure I will look at it ASAP 😸

**Comment by dlassalle** [comment]
I have these changes running livechat.viasatconnect.com right now and they look to be functioning well
Once we merge the other PRs, we also need to update the package.json here to be able to deploy to our servers with rc-apps cli tool

I'll let Prajval take a look, then we can start merging everything

**Comment by PrajvalRaval** [comment]
Tested the User typing indicator. LGTM 👌. Just suggesting the following two very minor additions for proper logging. 😄

File: lib/OnUserTypingHandler.ts
Comment (PrajvalRaval) [inline lib/OnUserTypingHandler.ts:53]: You can create an ENUM in the InfoLog file and log it here instead.

File: lib/OnUserTypingHandler.ts
Comment (PrajvalRaval) [inline lib/OnUserTypingHandler.ts]: I would suggest using a normal console.log(), with the following structure similar to this: console.log(Infolog.RECIEVED_TYPING_HANDLER_DATA, data)

File: lib/OnUserTypingHandler.ts
Comment (Shailesh351) [reply lib/OnUserTypingHandler.ts:53]: Normally we don't log success events. And also this typing event will be too frequent so better to not log. On fail it makes more sense. 😄

File: lib/OnUserTypingHandler.ts
Comment (Shailesh351) [reply lib/OnUserTypingHandler.ts]: It it not needed actually. I added it for debugging purpose and forgot to remove 😅.  Will remove it. 
Thanks 😄

Comment (Shailesh351) [inline lib/OnUserTypingHandler.ts]:
```suggestion
```

File: lib/OnUserTypingHandler.ts
Comment (PrajvalRaval) [reply lib/OnUserTypingHandler.ts:53]: Makes Sense 💯

**Review by PrajvalRaval — APPROVED** [review]

**Comment by Shailesh351** [comment]
@dlassalle Updated package.json dependency to point our apps-engine repo. We can merge 👍

**Pull request merged**
